ssl/statem/extensions_srvr.c

```diff
@@ -1635,6 +1673,33 @@ EXT_RETURN tls_construct_stoc_maxfragmentlen(SSL_CONNECTION *s, WPACKET *pkt,
     return EXT_RETURN_SENT;
 }
 
+EXT_RETURN tls_construct_stoc_record_size_limit(SSL_CONNECTION *s, WPACKET *pkt,
+                                             unsigned int context, X509 *x,
+                                             size_t chainidx)
+{
+    if (s->options & SSL_OP_NO_RECORD_SIZE_LIMIT_EXT)
+        return EXT_RETURN_NOT_SENT;
+
+    /* If the peer did not send a Record Size Limit. */
+    if (s->session->ext.peer_record_size_limit ==
+        TLSEXT_record_size_limit_UNSPECIFIED)
+        return EXT_RETURN_NOT_SENT;
```

Comment:
Shouldn't be necessary. We won't be called if we didn't receive a record size limit extension. Although possibly we want this as an `ossl_assert`??

```suggestion
    if (!ossl_assert(s->session->ext.peer_record_size_limit
                     != TLSEXT_record_size_limit_UNSPECIFIED)) {
        SSLfatal(s, SSL_AD_INTERNAL_ERROR, ERR_R_INTERNAL_ERROR);
        return EXT_RETURN_FAIL;
    }
```